src/main/java/io/cdap/plugin/gcp/bigtable/sink/BigtableSink.java

```diff
@@ -0,0 +1,223 @@
+    Mutation mutation = transformer.transform(record);
+    emitter.emit(new KeyValue<>(null, mutation));
+  }
+
+  private Configuration getConfiguration() {
+    Configuration conf = HBaseConfiguration.create();
+    BigtableConfiguration.configure(conf, config.getProject(), config.instance);
+    conf.set(BigtableOptionsFactory.BIGTABLE_RPC_TIMEOUT_MS_KEY, "60000");
+    conf.set(TableOutputFormat.OUTPUT_TABLE, config.table);
+    return conf;
+  }
+
+  private void validateInputSchema(Schema inputSchema) {
+    if (null == inputSchema.getField(config.keyAlias)) {
+      throw new InvalidConfigPropertyException(
+        String.format("Field '%s' declared as key alias does not exist in input schema", config.keyAlias),
+        BigtableSinkConfig.KEY_ALIAS
+      );
+    }
+    List<Schema.Field> fields = inputSchema.getFields();
+    if (null == fields || fields.isEmpty()) {
+      throw new InvalidStageException("Input schema should contain fields");
+    }
+    for (Schema.Field field : fields) {
+      Schema.Type fieldType = field.getSchema().isNullable() ?
+        field.getSchema().getNonNullable().getType() : field.getSchema().getType();
+      if (!SUPPORTED_FIELD_TYPES.contains(field.getSchema().getType())) {
+        String supportedTypes = SUPPORTED_FIELD_TYPES.stream()
+          .map(Enum::name)
+          .map(String::toLowerCase)
+          .collect(Collectors.joining(", "));
+        String errorMessage = String.format("Field '%s' is of unsupported type '%s'. Supported types are: %s.",
+                                            field.getName(), fieldType, supportedTypes);
+        throw new InvalidStageException(errorMessage);
+      }
+    }
+  }
+
+  private void createTable(Connection connection, TableName tableName) {
+    try (Admin admin = connection.getAdmin()) {
+      HTableDescriptor tableDescriptor = new HTableDescriptor(tableName);
+      tableDescriptor.addFamily(new HColumnDescriptor(config.columnFamily));
+      admin.createTable(tableDescriptor);
+    } catch (IOException e) {
+      throw new InvalidStageException(String.format("Failed to create table '%s' in Bigtable", tableName), e);
+    }
+  }
+
+  private void validateExistingTable(Connection connection, TableName tableName) {
+    try (Table table = connection.getTable(tableName)) {
+      if (!table.getTableDescriptor().hasFamily(Bytes.toBytes(config.columnFamily))) {
+        throw new InvalidConfigPropertyException(
+          String.format("Column family '%s' does not exist", config.columnFamily),
+          BigtableSinkConfig.COLUMN_FAMILY
+        );
+      }
+    } catch (IOException e) {
+      throw new InvalidStageException("Failed to connect to Bigtable", e);
+    }
+  }
+
+  private void emitLineage(BatchSinkContext context) {
+    Schema inputSchema = context.getInputSchema();
+    LineageRecorder lineageRecorder = new LineageRecorder(context, config.referenceName);
```

Comment:
should only do this if the input schema is not null

Reply:
fixed